### ISSUE TITLE
(tentative d') Ajout BU UJM

### ISSUE THREAD

**Issue by dominique-r** [lovasoa/ophirofox#125]
Bonjour, 

J'ai essayé de suivre les instructions pour ajouter "un nouveau partenaire Europresse"
mais je ne suis pas sur d'avoir fait correctement les choses (le déroulé n'est pas exactement le même qu'indiqué ...)
Il y a bien un patch proposé sur un fork
https://github.com/lovasoa/ophirofox/compare/master...dominique-r:ophirofox:patch-1

mais je ne le vois rien dans les pull-request 

Par ailleurs j'aurais deux questions : 
1/ sur le manifest.json que j'avais bidouillé en locale pour tester il y avait une entrée 
 "https://nouveau.europresse.com/*",
au niveau "optional_permissions": [
Elle me parait indispensable pour l'UJM compte tenu que nous ne sommes pas derrière un EZproxy, mais en accès sous controle de l'adresse IP
Je l'ai donc rajouté dans ma proposition de modif. 
A moins que ce soit l'entrée "webNavigation", qui joue se rôle ?

2/ il est indiqué "Conservez l'ordre alphabétique des universités." 
sauf que j'ai eu l'impression qu'il y avait plusieurs type d'ordre alphabétique mélangés : 
un qui ne tient pas compte des article et un qui en tient compte.
Ce qui après coup n'est pas le cas, c'est un ordre alpha qui ne tient pas compte des articles
et donc j'ai mal classé la nouvelle entrée ! : -\
... j'ai essayé de corriger ...

**Comment by lovasoa** [comment]
Très bien, vous avez tout fait comme il faut jusqu'ici. Il ne vous manque plus qu'à cliquer sur le gros bouton vert *create pull request*

![image](https://github.com/lovasoa/ophirofox/assets/552629/5851915c-4ab9-47d6-97a2-b71f890a5afa)


Si votre accès se fait directement sur `nouveau.europresse.com`, il n'y a rien à ajouter dans optional_permissions, le site principal d'europresse est déjà dans les permissions obligatoires (non optionnelles).

**Comment by lovasoa** [comment]
Vous pouvez continuer à travailler dans votre branche sur github (https://github.com/dominique-r/ophirofox/tree/patch-1) après avoir ouvert la pull request, cette dernière se mettra à jour automatiquement.

**Comment by dominique-r** [comment]
Bonjour,

Suite à la communication de la version de test de l'extension :
Okay pour la partie installation temporaire de l'extension.
Par contre au moment de la sélection de l'université dans les préférences de l'extension, 
je ne peux pas cocher l'option l'UJM.
Toutes les autres universités, oui mais Saint Etienne non  ...
?

Cordialement

**Comment by lovasoa** [comment]
Que se passe-t-il lorsque vous cliquez sur l'université ?

**Comment by dominique-r** [comment]
Rien !
je ne peux pas cocher le bouton radio.

**Comment by dominique-r** [comment]
Un message que je n'avais pas vu au niveau de la fenêtre about:debugging#/runtime/this-firefox

Détails de l’avertissement
Reading manifest: Warning processing background.type: An unexpected property was found in the WebExtension manifest.

**Comment by lovasoa** [comment]
> Rien !

D'accord. Je vais regarder ce qu'il se passe.

>  Reading manifest: Warning processing background.type: An unexpected property was found in the WebExtension manifest.

ça c'est juste parce que vous avez une ancienne version de firefox, je pense.

**Comment by dominique-r** [comment]
| parce que vous avez une ancienne version de firefox, je pense.
hummm ... je teste avec un FF Browser Extended Support Release 102.15.0esr (64bits)  - Firefox est à jour 
Mais si c'est sans lien avec le pb okay.

**Comment by lovasoa** [comment]
Oui, voilà. À propos de firefox 102, le pb [est décrit là](https://github.com/lovasoa/ophirofox/issues/103), il ne se produit pas avec les versions de firefox >= 105. C'est sans lien avec ce qui nous intéresse ici, je pense.

**Comment by dominique-r** [comment]
J'ai fait deux tests (recréer deux ophirofox.xpi localement à partir du ophirofox.xpi officiel )

le premier en modifiant la saisie des caractères accentués: \u00e9 au lieu de é et E simple)
même bugg

le second, j'ai ajouté (comme la première fois) 
"https://nouveau.europresse.com/*",
dans le bloc 
 "optional_permissions": [

Et là tout fonctionne !
....

**Comment by lovasoa** [comment]
Oui, je vois. Mais étant déjà dans les permissions obligatoires, j'aimerais bien que l'extension fonctionne sans avoir à le dupliquer dans les permissions optionnelles. Je vais faire une proposition de fix ajd.

**Comment by dominique-r** [comment]
C'est le fait que nous soyons avec une url https://nouveau.europresse.com/* (puisque contrôle par IP) qui a mis à jour ce pb.

**Comment by lovasoa** [comment]
J'ai intégré votre contribution. Est-ce que vous pourriez me confirmer que la dernière version fonctionne avec st étienne ?

**Comment by dominique-r** [comment]
Bonjour,
Je viens de l'installer sur un FF 115.2.1esr (64 bits) :
et il y a deux problèmes :
1/ pas d'affichage des noms d'établissements mais seulement de leur urls
exemple 
Autorisations facultatives pour des fonctionnalités supplémentaires :
Accéder à l’activité du navigateur pendant la navigation
Accéder à vos données pour https://nouveau-europresse-com.ezproxy.univ-catholille.fr
Accéder à vos données pour https://nouveau-europresse-com.ezproxy.upf.pf
etc 
et pas de module de recherche

2/ je n'ai pas retrouvé l'url de Saint Etienne

???

**Comment by lovasoa** [comment]
Je pense que vous n'avez pas installé la dernière version. La dernière est la 2.4.25857.16101

Vous pouvez voir le numéro de version installé sur firefox sur la page `about:addons`

![image](https://github.com/lovasoa/ophirofox/assets/552629/06db6aec-b347-43ec-8710-e29bb78cdf7d)

**Comment by dominique-r** [comment]
je viens de retenter, toujours sur un FF 115.2.1esr (64 bits)
et bien que je vienne d'installer la version 2.4.25859.12925,
je n'ai pas du tout le même affichage que vous :

![ophirofox_1](https://github.com/lovasoa/ophirofox/assets/4264006/7ad00fac-61d2-40d4-87e1-1438dca63190)
![ophirofox_2](https://github.com/lovasoa/ophirofox/assets/4264006/4c05d859-691b-4697-ba41-4a1b20761571)

**Comment by dominique-r** [comment]
Je viens de retester sur un autre PC : 
autre version de windows (Windows 10 Entreprise versus) et autre version de FF (102.15.1esr (64 bits) )
et là pas de problème !!

Mercredi, où j'aurais de nouveau accès à l'autre PC, j’essaierais la manip indiqué.

**Comment by dominique-r** [comment]
Test sur une config 
OS windows 7 professionnel service pack 1 (64 bits)
navigateur FF Extended support release 115.2.1esr (64 bits)  (FF à jour)

avec l'extension Release 2.4.25859.12925

Je ne voie rien au niveau de l'inspection de l'extension quand j'ouvre sa config (permission) et que je sélectionne ou désélectionne un bouton radio
![pb_config_Ophirofox_1](https://github.com/lovasoa/ophirofox/assets/4264006/55cb129b-bdd6-4d34-b511-9608eef6103d)
![pb_config_Ophirofox_2](https://github.com/lovasoa/ophirofox/assets/4264006/fe70aa05-821f-4f11-9960-6bcc5e3b0e54)

**Comment by lovasoa** [comment]
Et l'onglet "préférences" n'apparaît pas ?

**Comment by dominique-r** [comment]
non, je vois que les onglets Détails et Permissions

**Comment by lnoss** [comment]
@dominique-r, on notera qu'actuellement, la version de Firefox 115 sera la dernière version prise en charge pour les personnes utilisant Windows 7, Windows 8 ou Windows 8.1. Autrement dit, au quatrième quadrimestre de 2024, même si la prochaine version d'ESR pourrait contenir un correctif pour un potentiel bug, vous ne pourrez pas en profiter.

Je ne pense pas que des forks resteront compatibles avec Windows 7. Si la mise à jour du système d'exploitation n'est pas possible (mauvaise idée), il faudra peut-être regarder du côté de https://github.com/Skulltrail192/One-Core-API-Binaries
> This repositry contains binary releases of the One-Core-API project. These are compatible with Windows Server 2003 SP2, Windows XP SP3 and Windows XP x64 SP2.

C'est très pointilleux, mais en quelques mots : j'ai vu Firefox 115 ESR tourner sur Windows XP, je ne sais pas si c'est totalement utilisable et conseillé de rester... sur XP.

---

@lovasoa il y a peut-être une explication ici : https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/options_ui
> `browser_style`, deprecated in Manifest V3. Optional, defaulting to:
>- `true` in Manifest V2 and prior to Firefox 115 in Manifest V3.
>- `false` in Manifest V3 from Firefox 115.
>
> **Do not set browser_style to true: its not support in Manifest V3 from Firefox 118. See [Manifest V3 migration for browser_style](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Browser_styles#manifest_v3_migration).**

**Comment by lnoss** [comment]
Ai essayé avec la dernière version d'ESR, et la version [115.2.1 ESR](https://ftp.mozilla.org/pub/firefox/releases/115.2.1esr/win64/fr/) citée, impossible de reproduire le problème d'interface.

Les occurrences que je trouve sur des espaces de discussion laissent penser que cela peut être un problème de profil.
Saisir `about:profiles` dans la barre URL, redémarrer sans extensions puis avec pourrait éventuellement aider. Sinon, tenter de créer un autre profil ?

Documentation officielle liée : https://support.mozilla.org/fr/kb/gestionnaire-profils-creer-supprimer-changer-profils-firefox

**Comment by Write** [comment]
Je clos l'issue comme ceci a été fait dans #126 ! Merci à tous.
